help.go

```diff
@@ -12,12 +14,57 @@ func runHelp(args []string) {
 		mainUsage(os.Stdout)
 		return
 	}
-	fmt.Printf("specific help not yet implemented\n")
+	if len(args) != 1 {
+		fmt.Fprintf(os.Stderr, "gohack help %s: too many arguments\n", strings.Join(args, " "))
+		os.Exit(2)
+	}
+	t := template.Must(template.New("").Parse(commandHelpTemplate))
+	for _, c := range commands {
+		if c.Name() == args[0] {
+			if err := t.Execute(os.Stdout, c); err != nil {
+				fmt.Fprintf(os.Stderr, "cannot write output: %v", err)
+				os.Exit(1)
+			}
+			return
+		}
+	}
+	fmt.Fprintf(os.Stderr, "gohack help %s: unknown command\n", args[0])
+	os.Exit(2)
 }
 
 func mainUsage(f io.Writer) {
-	// TODO better than this
-	for _, cmd := range commands {
-		fmt.Fprintf(f, "%s\n", cmd.UsageLine)
+	t := template.Must(template.New("").Parse(mainHelpTemplate))
+	if err := t.Execute(f, commands); err != nil {
+		fmt.Fprintf(os.Stderr, "cannot write output: %v", err)
+		os.Exit(1)
 	}
 }
+
+var mainHelpTemplate = `
+The gohack command checks out Go module dependencies
+into a directory where they can be edited, and adjusts
+the go.mod file appropriately.
+
+Usage:
+
+	gohack <command> [arguments]
+
+The commands are:
+{{range .}}
+	{{.Name | printf "%-11s"}} {{.Short}}{{end}}
+
+Use "gohack help <command>" for more information about a command.
+
+As a special case,
+
+	gohack <module>
```

Comment:
Why? It seems to me like this complicates the tool's user interface. If a user wants shortcuts, they can always set up their own shell aliases however they want.

Reply:
Given that this is probably the most common thing for people to run, I thought it might be nice to keep some backward compatibility, but you're probably right. I'll remove it for the time being.